apps/namadillo/src/App/Transactions/TransactionCard.tsx

```diff
@@ -1,95 +1,244 @@
+import { TransactionHistory as TransactionHistoryType } from "@namada/indexer-client";
+import { shortenAddress } from "@namada/utils";
 import { TokenCurrency } from "App/Common/TokenCurrency";
-import { routes } from "App/routes";
-import { parseChainInfo } from "App/Transfer/common";
-import { chainRegistryAtom } from "atoms/integrations";
+import { AssetImage } from "App/Transfer/AssetImage";
+import { isShieldedAddress, isTransparentAddress } from "App/Transfer/common";
+import { indexerApiAtom } from "atoms/api";
+import { fetchBlockTimestampByHeight } from "atoms/balance/services";
+import { chainAssetsMapAtom } from "atoms/chain";
+import BigNumber from "bignumber.js";
 import clsx from "clsx";
 import { useAtomValue } from "jotai";
-import { GoIssueClosed, GoIssueTrackedBy, GoXCircle } from "react-icons/go";
-import { ImCheckmark } from "react-icons/im";
-import { generatePath, useNavigate } from "react-router-dom";
-import { twMerge } from "tailwind-merge";
+import { useEffect, useState } from "react";
+import { FaLock } from "react-icons/fa6";
 import {
-  ibcTransferStages,
-  IbcTransferTransactionData,
-  namadaTransferStages,
-  TransferTransactionData,
-} from "types";
-
-type TransactionCardProps = {
-  transaction: TransferTransactionData;
+  IoArrowBack,
+  IoArrowForward,
+  IoCheckmarkCircleOutline,
+  IoCloseCircleOutline,
+} from "react-icons/io5";
+import { twMerge } from "tailwind-merge";
+import { toDisplayAmount } from "utils";
+import keplrSvg from "../../integrations/assets/keplr.svg";
+
+type Tx = TransactionHistoryType;
+type Props = { tx: Tx };
+type RawDataSection = {
+  amount?: string;
+  sources?: Array<{ amount: string; owner: string }>;
+  targets?: Array<{ amount: string; owner: string }>;
 };
+const IBC_PREFIX = "ibc";
 
-const getTitle = (transferTransaction: TransferTransactionData): string => {
-  const { type } = transferTransaction;
+export function getToken(txn: Tx["tx"]): string | undefined {
+  const parsed = txn?.data ? JSON.parse(txn.data) : undefined;
+  if (!parsed) return undefined;
+  const sections = Array.isArray(parsed) ? parsed : [parsed];
 
-  if (Object.keys(namadaTransferStages).includes(type)) {
-    return "Transfer";
+  // return the first token found in sources or targets
+  for (const section of sections) {
+    if (section.sources?.length) {
+      return section.sources[0].token;
+    }
+    if (section.targets?.length) {
+      return section.targets[0].token;
+    }
   }
 
-  if (Object.keys(ibcTransferStages).includes(type)) {
-    return "Transfer IBC";
-  }
+  return undefined;
+}
 
-  return "";
+const titleFor = (kind: string | undefined, isReceived: boolean): string => {
+  if (!kind) return "Unknown";
+  if (isReceived) return "Receive";
+  if (kind.startsWith(IBC_PREFIX)) return "IBC Transfer";
+  if (kind === "transparentTransfer") return "Transparent Transfer";
+  if (kind === "shieldingTransfer" || kind === "unshieldingTransfer")
```

Comment:
This might be a picky detail, but maybe `kind === "unshieldingTransfer"` should map to `Unshielding Transfer` - I could see that being confusing / misleading. Would that break anything?